Fill in template contents for explainer repositories.

jyasskin opened the pull request:
@tabatkins, let me know what changes you suggest for the initial Bikeshed file.

jyasskin committed:
Fill in template contents for explainer repositories.

jyasskin committed:
Fix an invalid markup error.

jyasskin committed:
Remove more angle brackets...

jyasskin committed:
Add the required intro text.

cwilso commented on README.md:
```diff
@@ -1,2 +1,167 @@
+
+## Proponents
+
+- [Proponent team 1]
```
Should these have "todo" on them, or should the instructions say to look for brackets?

jyasskin replied:
I hope moving the "TODO: Fill in the whole explainer template" bit above this is sufficient.

cwilso commented on README.md:
```diff
@@ -1,2 +1,167 @@
+- [etc.]
+
+## Participate
+- [Issue tracker]
```
As above: todo here?

cwilso commented on README.md:
```diff
@@ -1,2 +1,167 @@
+
+## Introduction
+
+TODO: Fill this in using https://tag.w3.org/explainers/ as a reference.
```
It's not clear if this TODO is for the entire following sections or not?

jyasskin replied:
I've moved it up to the top of the explainer and reworded a bit so it's clear it covers the whole thing.

cwilso commented on README.md:
```diff
@@ -1,2 +1,167 @@
+
+[If spec work is in progress, link to the PR or draft of the spec.]
+
+## [Potential Solution 2]
```
This is possible, but relatively unlikely (that a second+ solution would be documented, right?  Perhaps we should remove this, but above say "there may be more than one potential solution section".

jyasskin replied:
Sounds good.

tabatkins commented on index.bs:
```diff
@@ -0,0 +1,24 @@
+
+For now, see the [explainer]([REPOSITORYURL]).
+
+See [https://speced.github.io/bikeshed/](https://speced.github.io/bikeshed/) to get started on your specificaton.
```
Drop in links to the other "how to Bikeshed" and "how to spec" links? So people have several jumping off points, besides just the (sometimes intimidating) Bikeshed docs.

jyasskin replied:
Done.

cwilso commented on README.md:
```diff
@@ -1,2 +1,167 @@
+<!-- In your initial explainer, you shouldn't be attached or appear attached to any of the potential
+solutions you describe below this. -->
+
+## [Potential Solution 1]
```
The potential solution(s) should be below the "tricky design choice" section below, but DEFINITELY after the Key Scenarios.

cwilso commented on README.md:
```diff
@@ -1,2 +1,167 @@
+and a brief overview of how the solution works.
+This should be no more than 1-2 paragraphs.]
+
+## Goals [or Motivating Use Cases, or Scenarios]
```
Not clear how this interacts with the Key Scenarios below?  Definitely need to make sure htis gets filled out, in depth, prior to the potential solution.

jyasskin replied:
Yeah. I took this directly from the [TAG's template](https://github.com/w3ctag/tag.w3.org/blob/main/explainers/template.md), but it's not clear. I think the idea was to list use cases here, and then have the Key Scenarios explain how the potential solutions solve the use cases. I'll try to make that clearer ... and then also send the TAG a patch.

cwilso reviewed (changes requested):
Two requested changes:
1) The instructions say "look for TODO" but really, each section needs to be filled out - either a bunch of sections need TODO in them, or just say "fill all this out".
2) Maybe I'm misunderstanding the Key Scenarios sections (or it needs retitling to "API interactions" or something), but it seems like it needs to come further up, where use cases and scenarios are listed.

tabatkins commented on index.bs:
```diff
@@ -0,0 +1,24 @@
+Repository: explainers-by-googlers/todo-your-repo-name
+URL: https://explainers-by-googlers.github.io/todo-your-repo-name
+Editor: TODO: Your Name, Google https://google.com, TODO@google.com
+!Tests: <a href=https://github.com/w3c/web-platform-tests/tree/master/todo-api-label>web-platform-tests todo-api-label/</a> (<a href=https://github.com/w3c/web-platform-tests/labels/todo-api-label>ongoing work</a>)
```
Is there a particular reason you're suggesting an explicitly-constructed WPT line, rather than using the Bikeshed's WPT features directly? This method won't, for example, fill in any test results.

Suggest instead a `WPT Path Prefix: TODO-API-LABEL` and `WPT Display: closed`, along with a `<wpt rest></wpt>` at the end of the body along with a short note explaining how to properly sprinkle `<wpt>` thruout the document.

jyasskin replied:
This is what `npx wicg` produced: https://github.com/WICG/starter-kit/blob/main/templates/index.bs. I've taken your suggestion, and I'll update that once this template is merged.

jyasskin committed:
Take Tab's suggestions.

jyasskin committed:
Try to fix Chris's comments.

jyasskin reviewed (commented):
Thanks for the reviews. How's this?

jyasskin commented on index.bs:
```diff
@@ -0,0 +1,40 @@
+and [https://speced.github.io/bikeshed/](https://speced.github.io/bikeshed/) to get started on your
+specificaton.
+
+<wpt rest></wpt>
```
Hmmm, @tabatkins this spelling produces an ["Attribute “rest” not allowed on element “details” at this point." error in spec-prod's markup validation](https://github.com/explainers-by-googlers/template/actions/runs/7023867643/job/19111358245?pr=1). Using `<wpt-rest>` as suggested in https://speced.github.io/bikeshed/#elementdef-wpt-rest fails Bikeshed with
```
FATAL ERROR: Couldn't find any tests with the path prefix 'TODO-API-LABEL'.
```

I could just remove this or comment this out, since the issue tells people what to do once they set their WPT Path Prefix. What do you think?

tabatkins replied:
Sorry, yes, `<wpt-rest>` is what I meant.

Re: the fatal error; go ahead and just ignore my suggestion for a `wpt-rest`, actually. It works just fine as-is; it (currently) won't complain at all about a nonsense path prefix as long as you don't actually use any wpt-related elements. But when that TODO is replaced with the actual path prefix, it'll start complaining about tests not being specified, which is good. (Tho I really need to work on the error message...)

jyasskin replied:
Removed. :)

jyasskin committed:
Remove the <wpt> bits in the content.

When the spec author fills in the WPT Path Prefix, it'll give them
errors that say to add <wpt> elements, which is as good as an inline
issue.
